### DISCUSSION TITLE
problems with routes or redirection

### DISCUSSION THREAD

**Discussion by devicnex** [vercel/next.js#65754] [Help]
### Summary

![image](https://github.com/vercel/next.js/assets/88792597/62510c47-cdbc-40b7-bb14-2086f42e4bf6)
![image](https://github.com/vercel/next.js/assets/88792597/9628472e-3957-4d68-8b66-cdcc4d57d516)

Olá, pessoal, estou com um problema de rotas, ai realizar a autenticação, não estou conseguindo direcionar para a tela inicial - painel, retorna no console o seguinte Error: NEXT_REDIRECT
e logo não realiza o direcionammento, isso tanto com o redirect ou router.push

**Comment by icyJoseph** [comment]
Hi,

The problem is that `redirect` throws... internally it will do something like `throw new REDIRECT_ERROR(/* ---  */)`


So by accident your try catch is, picking up on that, and preventing the runtime from receiving the redirect error, and doing its proper work. 

Read a bit more here, https://nextjs.org/learn/dashboard-app/error-handling#adding-trycatch-to-server-actions:

> Note how redirect is being called outside of the try/catch block. This is because redirect works by throwing an error, which would be caught by the catch block. To avoid this, you can call redirect after try/catch. redirect would only be reachable if try is successful.
